tabo-hdinsight/Azuredeploy.json

```diff
@@ -0,0 +1,555 @@
+              "addressPrefix": "[variables('TaboSubnetPrefix')]"
+            }
+          }
+        ]
+      }
+    },
+    {
+      "name": "[variables('TaboNetworkInterfaceName')]",
+      "type": "Microsoft.Network/networkInterfaces",
+      "apiVersion": "2015-06-15",
+      "location": "[variables('TaboLocation')]",
+      "dependsOn": [
+        "[concat('Microsoft.Network/virtualNetworks/', variables('TaboVirtualNetworkName'))]",
+        "[concat('Microsoft.Network/publicIpAddresses/', variables('TaboPublicIpAddressName'))]",
+        "[concat('Microsoft.Network/networkSecurityGroups/', variables('TaboNetworkSecurityGroupName'))]"
+      ],
+      "properties": {
+        "primary": true,
+        "ipConfigurations": [
+          {
+            "name": "ipconfig1",
+            "properties": {
+              "subnet": {
+                "id": "[variables('TaboSubnetRef')]"
+              },
+              "privateIPAllocationMethod": "Dynamic",
+              "publicIpAddress": {
+                "id": "[resourceId('Microsoft.Network/publicIpAddresses', variables('TaboPublicIpAddressName'))]"
+              }
+            }
+          }
+        ],
+        "networkSecurityGroup": {
+          "id": "[resourceId('Microsoft.Network/networkSecurityGroups', variables('TaboNetworkSecurityGroupName'))]"
+        }
+      }
+    },
+    {
+      "name": "[variables('TaboPublicIpAddressName')]",
+      "type": "Microsoft.Network/publicIPAddresses",
+      "apiVersion": "2015-06-15",
+      "location": "[variables('TaboLocation')]",
+      "properties": {
+        "publicIPAllocationMethod": "[variables('TaboPublicIpAddressType')]",
+        "dnsSettings": {
+          "domainNameLabel": "[variables('TaboVmdns')]"
+        }
+      }
+    },
+    {
+      "name": "[variables('TaboNetworkSecurityGroupName')]",
+      "type": "Microsoft.Network/networkSecurityGroups",
+      "apiVersion": "2015-06-15",
+      "location": "[variables('TaboLocation')]",
+      "properties": {
+        "securityRules": [
+          {
+            "name": "default-allow-rdp",
+            "properties": {
+              "priority": 1000,
+              "sourceAddressPrefix": "*",
+              "protocol": "TCP",
+              "destinationPortRange": "3389",
+              "access": "Allow",
+              "direction": "Inbound",
+              "sourcePortRange": "*",
+              "destinationAddressPrefix": "*"
+            }
+          },
+          {
+            "name": "default-allow-sql",
+            "properties": {
+              "priority": 1500,
+              "sourceAddressPrefix": "*",
+              "protocol": "Tcp",
+              "destinationPortRange": "1433",
+              "access": "Allow",
+              "direction": "Inbound",
+              "sourcePortRange": "*",
+              "destinationAddressPrefix": "*"
+            }
+          }
+        ]
+      }
+    },
+    {
+      "apiVersion": "[variables('TaboEhVersion')]",
```

Comment:
Please see #5 in best practice checklist: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md